### ISSUE TITLE
Switching import order causes compilation to fail with cryptic error

### ISSUE THREAD

**Issue by ChimeraCoder** [gopherjs/gopherjs#202]
I'm still not entirely certain I understand why this is occurring, but I'm almost certain it's a bug somewhere, as Go import order should not be impacting behavior.

The following code compiles without error

````go
package main

import (
    "strings"

    "github.com/gopherjs/gopherjs/js"
    gojson "github.com/ChimeraCoder/gojson"
)

type Pet struct {
    name string
}

func New(name string) *js.Object {
    return js.MakeWrapper(&Pet{name})
}

func (p *Pet) Name() string {
    return p.name
}

func (p *Pet) SetName(name string) {
    p.name = name
}

func main() {
    js.Global.Set("pet", map[string]interface{}{
        "New": New,
    })  
    input := strings.NewReader("asdf")

    if output, err := gojson.Generate(input, "Test", "FooPkg"); err != nil {
        panic(err)
    } else {
        print(output)
    }   

}
`````

However, the following code produces the cryptic error `test.go:27:2: invalid operation: js.Global (variable of type *github.com/gopherjs/gopherjs/js.Object) has no field or method Set`

````go
package main

import (
    "strings"

    gojson "github.com/ChimeraCoder/gojson"
    "github.com/gopherjs/gopherjs/js"
)

type Pet struct {
    name string
}

func New(name string) *js.Object {
    return js.MakeWrapper(&Pet{name})
}

func (p *Pet) Name() string {
    return p.name
}

func (p *Pet) SetName(name string) {
    p.name = name
}

func main() {
    js.Global.Set("pet", map[string]interface{}{
        "New": New,
    })  
    input := strings.NewReader("asdf")

    if output, err := gojson.Generate(input, "Test", "FooPkg"); err != nil {
        panic(err)
    } else {
        print(output)
    }   

}
````

This happens anytime the `gojson` package is imported before `gopherjs/js`, whether or not the qualified import name is used.

My original guess was that gopherjs is failing to compile  `github.com/ChimeraCoder/gojson`, but (A) this shouldn't affect the compilation of `gopherjs/js`, and (B) running `gopherjs build` inside the `gojson` directory does not produce any error.

**Comment by neelance** [comment]
That's definitely a bug. Thanks for reporting. I'll look into it.

**Comment by neelance** [comment]
Hmm, I just tried to reproduce this bug but I can't. Which OS are you using, which Go version and please try if `go get -u golang.org/x/tools/go/types` changes anything. Thanks! :-) We'll track this down.

**Comment by ChimeraCoder** [comment]
Updating `go/types` didn't help, so I pushed the branch `gopherjs_error_testing` (https://github.com/ChimeraCoder/gojson/tree/gopherjs_error_testing) to demonstrate the error. But then I thought to pull `HEAD` on gopherjs, and now it seems to be working. 

My gopherjs version couldn't have been too old, since it was on a relatively new machine, but whatever the issue was, it's apparently fixed now! (Unfortunately I did `go get -u` which does a fast-forward pull, so I can't check which version it was using before).
